### PR TITLE
Allow constant folding nodes that have missing optional inputs

### PR DESCRIPTION
**Pull request by hariharans29**
### Description
Some nodes may contain missing optional inputs that otherwise meet the criteria for constant folding. Currently, we do not constant fold these nodes. This change is to allow folding such nodes.

### Motivation and Context
Fix https://github.com/microsoft/onnxruntime/issues/15318

### CODE REVIEW

File: onnxruntime/core/graph/graph_utils.cc [hunk 1]
```diff
@@ -625,6 +625,11 @@ bool AllNodeInputsAreConstant(const Graph& graph, const Node& node, InitializedT
   }
 
   for (const auto* input_def : node.InputDefs()) {
+    // For optional node inputs which are missing, we can safely ignore them
+    if (input_def->Name() == "") {
```
Comment (pranavsharma) [inline onnxruntime/core/graph/graph_utils.cc:629]:
```suggestion
    if (input_def->Name().empty()) {
```

**Review by pranavsharma — DISMISSED** [review]

**Review by pranavsharma — APPROVED** [review]

**Comment by hariharans29** [comment]
/azp run 
Windows CPU CI Pipeline

**Comment by hariharans29** [comment]
/azp run 
iOS CI Pipeline

**Pull request merged**